ksqldb-engine/src/main/java/io/confluent/ksql/analyzer/RewrittenAnalysis.java

```diff
@@ -59,6 +59,10 @@ public RewrittenAnalysis(
     this.rewriter = Objects.requireNonNull(rewriter, "rewriter");
   }
 
+  public ImmutableAnalysis original() {
+    return original;
+  }
```

Comment:
I know, I don't like putting it back either.

The originals are needed for error messages, so that we get errors about `T.ID` not being in the projection, (which is what the user needs to add), and not `T_ID` which would result in an error if they added it.

Improving the way this handled is probably best left outside the scope of this PR though. :'(